### PR TITLE
Login: Require user to be signed out to get request password email 

### LINKED ISSUE #35253: Login/Reset Password: Wrong background when accessing reset-email route while logged in
**What happened**:
Screenshot from 2021-06-07 11-21-32

**What you expected to happen**:
To have the correct theme settings, alternatively not be able to access that route.

**How to reproduce it (as minimally and precisely as possible)**:
- Login
- Set theme to light
- Go directly to the URL `/user/password/send-reset-email`

**Anything else we need to know?**:

I can't imagine how this could happen naturally, but the access to the login page might cause some funky things to happen.

**Environment**:
- Grafana version: v8.1.0-pre
- OS Grafana is installed on: Ubuntu
- User OS & Browser: Chrome

### PR DESCRIPTION
**Pull request by tskarhed**
**What this PR does / why we need it**:
Restricts access to the reset-password-email route if you're logged in. Chances are you already know your password

**Which issue(s) this PR fixes**:
Closes #35253

### CODE REVIEW

**Comment by hugohaggmark** [comment]
Friendly reminder to add the correct milestone and backport label if this needs to be released in a patch release. Thank you!

File: pkg/middleware/auth.go [hunk 1]
```diff
@@ -161,6 +161,12 @@ func SnapshotPublicModeOrSignedIn(cfg *setting.Cfg) macaron.Handler {
 	}
 }
 
+func ReqNotSignedIn(c *models.ReqContext) {
+	if c.IsSignedIn && !c.IsAnonymous {
```
Comment (marefr) [inline pkg/middleware/auth.go:165]: I believe `c.IsSignedIn` is false when `c.IsAnonymous` is true according to [code](https://github.com/grafana/grafana/blob/75ff02a9fb6c06664171cde75ddf87336a9f1886/pkg/services/contexthandler/contexthandler.go#L144-L146) so think the following should be enough:

```suggestion
	if c.IsSignedIn {
```

**Review by marefr — CHANGES_REQUESTED** [review]
Looks good I think. Suggested some changes.

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**